aws/src/main/java/org/apache/iceberg/aws/s3/S3OutputStream.java

```diff
@@ -218,41 +241,50 @@ private void uploadParts() {
 
     stagingFiles.stream()
         // do not upload the file currently being written
-        .filter(f -> closed || !f.equals(currentStagingFile))
+        .filter(f -> closed || !f.file().equals(currentStagingFile))
         // do not upload any files that have already been processed
-        .filter(Predicates.not(multiPartMap::containsKey))
-        .forEach(f -> {
+        .filter(Predicates.not(f -> multiPartMap.containsKey(f.file())))
+        .forEach(fileAndDigest -> {
+          File file = fileAndDigest.file();
```

Comment:
I think we can name this variable `f` to avoid name changing in subsequent lines

Reply:
using `f` would lead to `LocalVariableName` checkstyle failure. Previous discussion here https://github.com/apache/iceberg/pull/3813#discussion_r788252447

Reply:
I'd rather suppress the checkstyle rule and have fewer changes.

Reply:
That's fine with me. Trying to understand the rational though. Is it to avoid merge conflicts for folks that may be doing a cherry-pick?

Reply:
To avoid conflicts.